src/bindings/c/include/openvino/c/ov_model.h

```diff
@@ -0,0 +1,148 @@
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param output_nodes A pointer to the ov_output_nodes.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_outputs(const ov_model_t* model, ov_output_node_list_t* output_nodes);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param input_nodes A pointer to the ov_input_nodes.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_inputs(const ov_model_t* model, ov_output_node_list_t* input_nodes);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param tensor_name input tensor name (char *).
+ * @param input_node A pointer to the ov_output_const_node_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_input_by_name(const ov_model_t* model, const char* tensor_name, ov_output_const_node_t** input_node);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param index input tensor index.
+ * @param input_node A pointer to the ov_input_node_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_input_by_index(const ov_model_t* model, const size_t index, ov_output_const_node_t** input_node);
+
+/**
+ * @brief Returns true if any of the op's defined in the model contains partial shape.
+ * @param model A pointer to the ov_model_t.
+ */
+OPENVINO_C_API(bool) ov_model_is_dynamic(const ov_model_t* model);
+
+/**
+ * @brief Do reshape in model with a list of <name, partial shape>.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param tensor_names input tensor name (char *) list.
+ * @param partialShape A PartialShape list.
+ * @param cnt The item count in the list.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_reshape(const ov_model_t* model,
+                 const char* tensor_names[],
+                 const ov_partial_shape_t* partial_shapes[],
+                 size_t cnt);
+
+/**
+ * @brief Do reshape in model with partial shape for a specified name.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param tensor_name input tensor name (char *).
+ * @param partialShape A PartialShape.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_reshape_input_by_name(const ov_model_t* model,
+                               const char* tensor_name,
+                               const ov_partial_shape_t* partial_shape);
+
+/**
+ * @brief Do reshape in model for one node(port 0).
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param partialShape A PartialShape.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_reshape_one_input(const ov_model_t* model, const ov_partial_shape_t* partial_shape);
```

Comment:
ov_model_reshape_one_input -> ov_model_reshape_single_input ?

Reply:
The name is from Ilya.C, maybe we can update it.

Reply:
ok